src/veil/lib/veil/hasher/bcrypt.rb

```diff
@@ -0,0 +1,58 @@
+require "veil/hasher/base"
+require "securerandom"
+require "bcrypt"
+
+module Veil
+  class Hasher
+    class BCrypt < Base
+      attr_reader :secret, :salt
+
+      # Create a new BCrypt
+      #
+      # @param [Hash] opts
+      #   a hash of options to pass to the constructor
+      def initialize(opts = {})
+        if opts[:secret] && opts[:salt]
+          if ::BCrypt::Engine.valid_secret?(opts[:secret]) && ::BCrypt::Engine.valid_salt?(opts[:salt])
+            @secret = opts.delete(:secret)
+            @salt = opts.delete(:salt)
+          elsif ::BCrypt::Engine.valid_secret?(opts[:secret])
+            raise Veil::InvalidSalt.new("#{opts[:salt]} is not valid salt")
+          else
+            raise Veil::InvalidSecret.new("#{opts[:secret]} is not valid secret")
+          end
+        else
+          # BCrypt is capped at 55 character input:
```

Comment:
Is this comment correct?  `SecureRanom.hex(53)` will return 106 characters since the 53 refers to bytes and each byte will be represented by two hexadecimal digits in the returned string.